### ISSUE TITLE
Errors on downloading

### ISSUE THREAD

**Issue by gkeegan** [TeamNewPipe/NewPipe#2360]
- [X] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them. 
- [X] I checked if the issue/feature exists in the latest version.
 - [X] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.

When downloading, the video kept going to --.-- mb/s, so I had to keep pausing and starting to get it to keep downloading. After a few times I got a crash.

## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Language:__ GB
* __Service:__ none
* __Version:__ 0.16.0 (this is on latest commits from dev branch as of May 22)
* __OS:__ Linux HONOR/BND-L24/HWBND-H:8.0.0/HONORBND-L24/344(C567):user/release-keys 9 - 28

<details><summary><b>Crash log</b></summary><p>

```
java.lang.IndexOutOfBoundsException: Index: 2, Size: 0
	at java.util.ArrayList.set(ArrayList.java:453)
	at us.shandian.giga.get.DownloadMission.setThreadBytePosition(DownloadMission.java:223)
	at us.shandian.giga.get.DownloadRunnable.run(DownloadRunnable.java:139)

```
</p></details>
<hr>

**Comment by gkeegan** [comment]
I also just noticed something. When it is on --.-- mb/s, it is still downloading, but the progress bar/percent doesn't change. I believe this is related to this issue.

So when the video is on --.-- mb/s for a while, pausing then starting causes the video to finish downloading almost instantly.

**Comment by AmirMukhtar16** [comment]
I am getting the same issue and in my case, it stuck on the post processing on. Let me know how to resolve the issue please.

**Comment by kapodamy** [comment]
Is a tricky bug, this happens when the download thread is paused and almost done.

**Comment by gkeegan** [comment]
I believe this was fixed by #2149, as I have yet to notice it since I've been using a build with that in it, so I will close this issue.

**Comment by kapodamy** [comment]
fixed in #2440

**Comment by gkeegan** [comment]
Great work @kapodamy! I'll close once that PR is merged.
